docs/admin/healthcheck.md

```diff
@@ -0,0 +1,248 @@
+traffic between Coder and workspace agents. Coder periodically queries the
+health of its configured DERP servers and may return one or more of the
+following:
+
+### <a name="EDERP01">EDERP01: DERP Node Uses Websocket</a>
+
+**Problem:** When Coder attempts to establish a connection to one or more DERP
+servers, it sends a specific `Upgrade: derp` HTTP header. Some load balancers
+may block this header, in which case Coder will fall back to
+`Upgrade: websocket`.
+
+This is not necessarily a fatal error, but a possible indication of a
+misconfigured reverse HTTP proxy. Additionally, while workspace users should
+still be able to reach their workspaces, connection performance may be degraded.
+
+> [!NOTE] This may also be shown if you have
+> [forced websocket connections for DERP](../cli/server.md#--derp-force-websockets).
+
+**Solution:** ensure that any configured reverse proxy does not strip the
+`Upgrade: derp` header.
+
+### <a name="EDERP02">EDERP02: One or more DERP nodes are unhealthy</a>
+
+**Problem:** This is shown if Coder is unable to reach one or more configured
+DERP servers. Clients will fall back to use the remaining DERP servers, but
+performance may be impacted for clients closest to the unhealthy DERP server.
+
+**Solution:** Ensure that the DERP server is available and reachable over the
+network on port 443, for example:
+
+```shell
+curl -v "https://coder.company.com:443/derp"
+# Expected output:
+# *   Trying XXX.XXX.XXX.XXX:443
+# * Connected to https://coder.company.com (XXX.XXX.XXX.XXX) port 443 (#0)
+# DERP requires connection upgrade
+```
+
+## Websocket
+
+Coder makes heavy use of [WebSockets](https://datatracker.ietf.org/doc/rfc6455/)
+for long-lived connections:
+
+- Between users interacting with Coder's Web UI (for example, the built-in
+  terminal, or VSCode Web),
+- Between workspace agents and `coderd`,
+- Between Coder [workspace proxies](../admin/workspace-proxies.md) and `coderd`.
+
+Any issues causing failures to establish WebSocket connections will result in
+**severe** impairment of functionality for users. To validate this
+functionality, Coder will periodically attempt to establish a WebSocket
+connection with itself using the configured [Access URL](#access-url), send a
+message over the connection, and attempt to read back that same message.
+
+### <a name="EWS01">EWS01: Failed to establish a WebSocket connection</a>
+
+**Problem:** Coder was unable to establish a WebSocket connection over its own
+Access URL.
+
+**Solution:** There are multiple possible causes of this problem:
+
+1. Ensure that Coder's configured Access URL can be reached from the server
+   running Coder, using standard troubleshooting tools like `curl`:
+
+   ```shell
+   curl -v "https://coder.company.com:443/"
+   ```
+
+2. Ensure that any reverse proxy that is sitting in front of Coder's configured
+   access URL is not stripping the HTTP header `Upgrade: websocket`.
```

Comment:
Ensure that any reverse proxy that is serving Coder's configured access URL allows connection upgrade with the header `Upgrade: websocket`

Reply:
:+1: